### ISSUE TITLE
Error linking new environment (stack_rollback)

### ISSUE THREAD

**Issue by txlnghrns** [awslabs/aws-dataall#148]
Using version 1.1.1 of code - deployment went smoothly across tooling and deployment accounts. However, when linking a new environment, running into issues.

Issue 1: Had to manually add a line in Trust Policy to allow for dataall-dev-graphql-role to do stsAssume

Issue 2: Resolved (#147) unchecked Quicksight, else linking new environment will fail. Missed this step in the docs.

Issue 3: Stack rollback due to errors related to invalid principals. Log (scrubbed) attached.
[linkerr.MD](https://github.com/awslabs/aws-dataall/files/9657817/linkerr.MD)

Please let me know if there's guidance on this error message? I was able to recreate it a few times in same account.

Thank you team for your hard work on this project.

**Comment by txlnghrns** [comment]
Update: disabling ALL features produced a successful environment link. I would still love guidance on the above, as the features are a great addition. Thank you!

**Comment by dlpzx** [comment]
Hi @txlnghrns, thanks for opening the issue. 
Regarding the first error, could you share the change that you applied on the graphql role?

You are absolutely right, we can enhance the code by providing checks on the features that enabled/disabled. When it comes to Quicksight, not all actions can be performed via API. This was the case (until recently) for subscribing or checking to Quicksight subscriptions. Now we can use the following: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/quicksight.html#QuickSight.Client.describe_account_subscription. I have opened an enhancement issue for this #150 

The other issue is related to SageMaker domain, it is a reported bug, we are looking into it, but thank you for pointing it out so we don't forget.

**Comment by txlnghrns** [comment]
@dlpzx thanks for letting me know; as always, incredibly thankful to you for working so hard on this great project. Here's the block that I edited to get stsAssume to work. Note: I also can't find where I can enter ExternalID to input, so I disabled it.

`{
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::333333333:role/dataall-dev-graphql-role"
            },
          `

**Comment by txlnghrns** [comment]
Hi @dlpzx I believe the issue I had was being unable to click on the "External ID" in the UI on the Link Environments screen. Clicking on it produces an error. I am reading that clicking on the External ID button (which currently errors on me) would produce an External ID that I would then use to input into the parameters when running that CF stack?

**Comment by dlpzx** [comment]
Exactly, the external ID is stored in AWS Secrets Manager in the infrastructure account. It is a unique identifier of the account that prevents other accounts and roles to assume the pivot role (recommended way of assuming roles as explained in the [documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html)).

Do you still get the error of not getting the external ID from the UI?

**Comment by txlnghrns** [comment]
I sure do. This is the error.
<img width="921" alt="Screen Shot 2022-10-06 at 8 05 40 AM" src="https://user-images.githubusercontent.com/113570075/194320435-0c789c86-d23c-4503-8127-9e4e7268df70.png">

**Comment by dlpzx** [comment]
Hi @txlnghrns, your error is unrelated to the backend-frontend of data.all and its application logic. It seems to be caused by your browser which in this case I think it is Safari. We are copying information to the Clipboard and it looks like the interactions between your browser and the Clipboard are blocked. Until we find an alternative and dive into the issue, I would recommend you to first try to open the UI with a different browser. Both parameters, the externalID and the Pivot role name are stored in AWS Secrets Manager in the infra accounts, so you can also copy them from there in case everything fails.

**Comment by txlnghrns** [comment]
Hi @dlpzx thanks, that did the trick. Interesting as the clipboard copy worked for the Pivot role name, but not for External Id in Safari, so I made an assumption it was doing something other than clipboard copy. I should've noted the icon used are always for clipboard copy. Thanks again!
